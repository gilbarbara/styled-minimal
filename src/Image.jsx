import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';

import Box from './Box';

const Image = styled(Box)(props => {
  const { maxWidth } = props;

  return css`
    max-width: ${maxWidth || '100%'};
  `;
});

Image.displayName = 'Image';

Image.propTypes = {
  alt: PropTypes.string.isRequired,
  as: PropTypes.string,
  src: PropTypes.string.isRequired,
  ...Box.basePropTypes,
};

Image.defaultProps = {
  as: 'img',
};

export default Image;
