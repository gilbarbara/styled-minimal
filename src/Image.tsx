import * as React from 'react';
import styled from 'styled-components';

import { baseStyles, sx, system } from './utils/system';

import { ImageProps } from './types';

export const StyledImage = styled('img')<any>(
  {
    ...baseStyles,
    maxWidth: '100%',
  },
  sx,
  system,
);

export const Image = React.forwardRef<HTMLImageElement, ImageProps>((props, ref) => (
  <StyledImage ref={ref} {...props} />
));
