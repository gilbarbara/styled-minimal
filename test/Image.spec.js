import React from 'react';
import { mount } from 'enzyme';

import Image from '../src/Image';

describe('Image', () => {
  const wrapper = mount(
    <Image src="https://assets.imgix.net/unsplash/moon.jpg" alt="Moon" />
  );

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
