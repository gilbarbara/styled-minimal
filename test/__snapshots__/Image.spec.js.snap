// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render properly 1`] = `
.c0 {
  box-sizing: border-box;
  max-width: 100%;
}

<Image
  alt="Moon"
  as="img"
  src="https://assets.imgix.net/unsplash/moon.jpg"
>
  <img
    alt="Moon"
    className="c0"
    src="https://assets.imgix.net/unsplash/moon.jpg"
  />
</Image>
`;
